app/src/androidTest/java/com/github/wnder/UserTesting.java

```diff
@@ -0,0 +1,107 @@
+package com.github.wnder;
+
+import android.net.Uri;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QuerySnapshot;
+import com.google.firebase.storage.FirebaseStorage;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.google.android.gms.tasks.Tasks.await;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(JUnit4.class)
+public class UserTesting {
+
+    @Test
+    public void getNewPictureForSignedInUserWorks() throws ExecutionException, InterruptedException, TimeoutException {
+        SignedInUser u = new SignedInUser("testUser", Uri.parse("android.resource://com.github.wnder/" + R.raw.ladiag));
+        String pic = u.getNewPicture();
+        FirebaseFirestore storage = FirebaseFirestore.getInstance();
+
+        //Check that it is not in user's uploaded and guessed pictures
+        Set<String> upAdownPics = new HashSet<>();
+        Task<DocumentSnapshot> task = storage.collection("users").document("testUser").get().addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>() {
+            @Override
+            public void onSuccess(DocumentSnapshot documentSnapshot) {
```

Comment:
As the `addOnSuccessListener` make it clear what will be in the following function, you could use a lambda here:
```
addOnSuccessListener((DocumentSnapshot documentSnapshot) -> {
// onSuccess content goes here
});
```
(Same goes for all other onSuccessListener in the PR)